### ISSUE TITLE
Minetown food shop doesn't convert to health food store for monks

### ISSUE THREAD

**Issue by elunna** [k21971/EvilHack#104]
shknam.c seems to have the code for this but after visiting minetown 3 times with 3 different monks, it always resulted in a standard deli.

shknam.c lines 501-505:
```
    if (nlp == shkfoods && In_mines(&u.uz) && Role_if(PM_MONK)
        && (sptr = Is_special(&u.uz)) != 0 && sptr->flags.town) {
        /* special-case override for minetown food store for monks */
        nlp = shkhealthfoods;
    }
```

**Comment by k21971** [comment]
See https://github.com/elunna/hackem/issues/54

**Comment by k21971** [comment]
Fixed via commit 14eb6b6.
